src/gradletestrunnersmoketest.sh

```diff
@@ -18,14 +18,25 @@ gradletestrunnersmoketest () {
   ./gradlew spotlessCheck
   ./gradlew spotbugsMain
 
-  echo "Running test suite"
-  ./gradlew runTest --args="suites/PRSmokeTests.json tmp/TestRunnerResults"
+  local outputDir="/tmp/TestRunnerResults"
+  echo "Output directory set to: $outputDir"
+  
+  echo "Running tests"
+  ./gradlew runTest --args="suites/PRSmokeTests.json $outputDir" ||
+    (echo "Running tests FAILED" &&
+    ./gradlew uploadResults --args="BroadJadeDev.json $outputDir" &&
+    return 1)
```

Comment:
I'm guessing the exit caused an issue in the action? Also, are the parentheses strictly necessary (I do think they add clarity, but just curious)?

Reply:
Yes, the parentheses are strictly necessary. Otherwise, we always execute the return 1 and never make it to the next command, regardless of whether runTest fails or succeeds.

For example:
```
marikomedlock-macbookpro:datarepo-clienttests marikomedlock$ true || echo "after OR" && echo "after AND" && echo "after second AND"
after AND
after second AND
marikomedlock-macbookpro:datarepo-clienttests marikomedlock$ false || echo "after OR" && echo "after AND" && echo "after second AND"
after OR
after AND
after second AND
```